src/gsy_e/models/strategy/future/strategy.py

```diff
@@ -0,0 +1,258 @@
+        return FutureTemplateStrategyOfferUpdater(
+            initial_rate=initial_selling_rate,
+            final_rate=final_selling_rate,
+            fit_to_limit=True,
+            energy_rate_change_per_update=None,
+            update_interval=duration(minutes=_update_interval),
+            rate_limit_object=max)
+    return TemplateStrategyUpdaterInterface()
+
+
+class FutureMarketStrategy(FutureMarketStrategyInterface):
+    """Manages bid/offer trading strategy for the future markets, for a single asset."""
+    def __init__(self, asset_type: AssetType,
+                 initial_buying_rate: float, final_buying_rate: float,
+                 initial_selling_rate: float, final_selling_rate: float):
+        """
+        Args:
+            initial_buying_rate: Initial rate of the future bids
+            final_buying_rate: Final rate of the future bids
+            initial_selling_rate: Initial rate of the future offers
+            final_selling_rate: Final rate of the future offers
+        """
+        super().__init__()
+        self._offer_updater = future_strategy_offer_updater_factory(
+            initial_selling_rate, final_selling_rate, asset_type
+        )
+
+        self._bid_updater = future_strategy_bid_updater_factory(
+            initial_buying_rate, final_buying_rate, asset_type
+        )
+
+    def event_market_cycle(self, strategy: "BidEnabledStrategy") -> None:
+        """
+        Should be called by the event_market_cycle of the asset strategy class, posts
+        settlement bids and offers on markets that do not have posted bids and offers yet
+        Args:
+            strategy: Strategy object of the asset
+
+        Returns: None
+
+        """
+        if not strategy.area.future_markets:
+            return
+        self._bid_updater.update_and_populate_price_settings(strategy.area)
+        self._offer_updater.update_and_populate_price_settings(strategy.area)
+        for time_slot in strategy.area.future_markets.market_time_slots:
+            if strategy.asset_type == AssetType.CONSUMER:
+                required_energy_kWh = strategy.state.get_energy_requirement_Wh(time_slot) / 1000.0
+                self._post_consumer_first_bid(strategy, time_slot, required_energy_kWh)
+            elif strategy.asset_type == AssetType.PRODUCER:
+                available_energy_kWh = strategy.state.get_available_energy_kWh(time_slot)
+                self._post_producer_first_offer(strategy, time_slot, available_energy_kWh)
+            elif strategy.asset_type == AssetType.PROSUMER:
+                available_energy_sell_kWh = strategy.get_available_energy_to_sell_kWh(time_slot)
+                available_energy_buy_kWh = strategy.get_available_energy_to_buy_kWh(time_slot)
+                self._post_producer_first_offer(strategy, time_slot, available_energy_sell_kWh)
+                self._post_consumer_first_bid(strategy, time_slot, available_energy_buy_kWh)
```

Comment:
Assigning a new variable is only for better reading of the code, right? Or is there another reason to not pass the `strategy.get_available_energy_to_sell_kWh(time_slot)` and `strategy.get_available_energy_to_buy_kWh(time_slot)` into the `_post_producer_first_offer` and `_post_consumer_first_bid` directly?

Reply:
Only for readability purposes. If you prefer it that way I can get rid of the intermediate variables though.

Reply:
ok, then leave it as is.